src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/legacy/CancelVisitTest.kt

```diff
@@ -281,13 +290,230 @@ class CancelVisitTest(@Autowired private val objectMapper: ObjectMapper) : Integ
     verify(telemetryClient, times(0)).trackEvent(eq("visit.cancelled-domain-event"), any(), isNull())
   }
 
+  /**
+   * the check for cancellations does not calculate time - only dates.
+   */
+  @Test
+  fun `cancel expired visit on same day as allowed day does not return error`() {
+    val outcomeDto = OutcomeDto(
+      OutcomeStatus.CANCELLATION,
+      "No longer joining."
+    )
+    // Given
+    val visitStart = LocalDateTime.now().minusDays(visitCancellationDayLimit).truncatedTo(ChronoUnit.DAYS).withHour(1)
+    val visit = visitEntityHelper.create(visitStatus = BOOKED, visitStart = visitStart)
+
+    // When
+    val responseSpec = webTestClient.patch().uri("/visits/${visit.reference}/cancel")
+      .headers(setAuthorisation(roles = listOf("ROLE_VISIT_SCHEDULER")))
+      .body(
+        BodyInserters.fromValue(outcomeDto)
+      )
+      .exchange()
+
+    // Then
+    responseSpec.expectStatus().isOk
+      .expectBody()
+      .jsonPath("$.visitStatus").isEqualTo(VisitStatus.CANCELLED.name)
+      .jsonPath("$.outcomeStatus").isEqualTo(OutcomeStatus.CANCELLATION.name)
+      .returnResult()
+
+    // And
+    verify(telemetryClient, times(1)).trackEvent(eq("visit-cancelled"), any(), isNull())
```

Comment:
not sure you have to test for events as other test do that,... alll your doing here is testing that you can cancel.. so status is enough

Reply:
Agreed. I will refactor to use the assertVisitCancellation method as suggested below and remove some asserts that are not needed.